From: Ray Strode <rstrode@redhat.com>
Date: Fri, 9 Feb 2024 09:02:25 -0500
Subject: authPrompt: Fade out cancel button after verification

Right now we make the cancel button non-reactive when
verification is complete, but it still disrupts the
login transition because in many cases most of the other
screen elements have been faded out.

This commit fades it out as well.
---
 js/gdm/authPrompt.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/js/gdm/authPrompt.js b/js/gdm/authPrompt.js
index 96338b1..d5cf9b9 100644
--- a/js/gdm/authPrompt.js
+++ b/js/gdm/authPrompt.js
@@ -541,6 +541,11 @@ export const AuthPrompt = GObject.registerClass({
         this.verificationStatus = AuthPromptStatus.VERIFICATION_SUCCEEDED;
         this.cancelButton.reactive = false;
         this.cancelButton.can_focus = false;
+        this.cancelButton.ease({
+            opacity: 0,
+            duration: MESSAGE_FADE_OUT_ANIMATION_TIME,
+            mode: Clutter.AnimationMode.EASE_OUT_QUAD,
+        });
     }
 
     _onReset() {
